lib/active_merchant/billing/gateways/authorize_net_cim.rb

```diff
@@ -864,10 +864,12 @@ def commit(action, request)
         response_params['direct_response'] = parse_direct_response(response_params['direct_response']) if response_params['direct_response']
         transaction_id = response_params['direct_response']['transaction_id'] if response_params['direct_response']
 
-        Response.new(success, message, response_params,
-          :test => test_mode,
-          :authorization => transaction_id || response_params['customer_profile_id'] || (response_params['profile'] ? response_params['profile']['customer_profile_id'] : nil)
-        )
+        response_options = {}
+        response_options[:test] = test_mode
+        response_options[:authorization] = transaction_id || response_params['customer_profile_id'] || (response_params['profile'] ? response_params['profile']['customer_profile_id'] : nil)
+        response_options[:error_code] = response_params['messages']['message']['code'] unless success
```

Comment:
@aprofeit best I found in the docs is [here](http://developer.authorize.net/api/reference/index.html#payment-transactions-charge-a-credit-card).

It mentions them as part of the response (see "Response" tab of "Charge a Credit Card"), yet doesn't state that it will always be there. 

<b>But</b>  for another param, `cavvResultCode` it says "Blank or not present = CAVV not validated.". 

So you'd think they would mention it if this was sometimes blank or not present.